aten/src/ATen/Utils.h

```diff
@@ -137,16 +138,14 @@ inline int64_t prod_intlist(ArrayRef<int64_t> list) {
  * See Note [Thread-safety and Generators]
  */
 template <typename T>
-static inline T * check_generator_with_default(Generator * expr, Generator * defaultValue) {
+static inline T * check_generator_with_default(Generator * expr, std::unique_ptr<T>& defaultValue) {
```

Comment:
If you end up agreeing with me about the pointer lifetime bits, you should undo this change